### PR TITLE
[CORE-API] Create User entity, User and Auth routes, controllers and services

### PR DESCRIPTION
**Pull request by viniciusgiles**
Solves #3 

### Added packages
- `class-validator`

### Notes
- Warning: *run migrations after merge!*

### CODE REVIEW

File: core-api/src/routes/auth.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Router } from "express";
+import { AuthController } from "../controllers/auth";
+
+function AuthRouter(): Router {
+    const router = Router();
+
+    router.post("/signup", AuthController.signUp);
+    // router.post("/login", AuthController.logIn);
```
Comment (eduardosm7) [inline core-api/src/routes/auth.ts]: Was it meant to be commented?

Comment (eduardosm7) [reply core-api/src/routes/auth.ts]: I see, login in AuthService is not implemented

File: core-api/src/controllers/auth.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { Request, Response } from "express";
+import { AuthService } from "../services/auth";
+import { HttpStatus } from "../utils/httpStatus";
+import { ServerError, ErrorCode } from "../utils/serverError";
+import Messages from "../utils/messages";
+
+export class AuthController {
+    public static async signUp(req: Request, res: Response) {
```
Comment (aacgn) [inline core-api/src/controllers/auth.ts]: This method must have a type declaration of void

Comment (aacgn) [reply core-api/src/controllers/auth.ts]: Also methods void don't have a return, even being a "return;"

File: core-api/src/controllers/auth.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { Request, Response } from "express";
+import { AuthService } from "../services/auth";
+import { HttpStatus } from "../utils/httpStatus";
+import { ServerError, ErrorCode } from "../utils/serverError";
+import Messages from "../utils/messages";
+
+export class AuthController {
```
Comment (aacgn) [inline core-api/src/controllers/auth.ts]: It's important when we are creating a lot of files with the same name (auth.ts) add a middle-extension. Imagine that we trying access this file by the name, will appear a lot of files and you probably will waste a lot of time searching for this specific one. Please, rename it for auth.controller.ts.

File: core-api/src/controllers/auth.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    public static async signUp(req: Request, res: Response) {
+        const { name, email, password } = req.body;
+
+        if (!name || !email || !password) {
```
Comment (aacgn) [inline core-api/src/controllers/auth.ts]: It's a better approach for performance and readability to start using a if (name && email && password) with the newUser and else with the error.

File: core-api/src/controllers/auth.ts (hunk 3 above)
Comment (aacgn) [reply core-api/src/controllers/auth.ts]: Please, include the middle-extension in the other files too.

**Review by aacgn — APPROVED** [review]

**Pull request merged**